au4/src/trackedit/internal/au3/au3trackeditproject.cpp

```diff
@@ -67,9 +84,54 @@ muse::async::NotifyList<au::trackedit::Track> Au3TrackeditProject::trackList() c
     return au4tracks;
 }
 
+static std::string eventTypeToString(const TrackListEvent& e)
+{
+    switch (e.mType) {
+    case TrackListEvent::SELECTION_CHANGE: return "SELECTION_CHANGE";
+    case TrackListEvent::TRACK_DATA_CHANGE: return "TRACK_DATA_CHANGE";
+    case TrackListEvent::PERMUTED: return "PERMUTED";
+    case TrackListEvent::RESIZING: return "RESIZING";
+    case TrackListEvent::ADDITION: return "ADDITION";
+    case TrackListEvent::DELETION: return e.mExtra ? "REPLACING" : "DELETION";
+    }
+}
+
+void Au3TrackeditProject::onTrackListEvent(const TrackListEvent& e)
+{
+    TrackId trackId = -1;
+    auto track = e.mpTrack.lock();
+    if (track) {
+        trackId = DomConverter::trackId(track->GetId());
+    }
+    LOGD() << "trackId: " << trackId << ", type: " << eventTypeToString(e);
+
+    switch (e.mType) {
+    case TrackListEvent::DELETION: {
+        if (e.mExtra == 1) {
+            m_impl->trackReplacing = true;
```

Comment:
Don't we need to reset this flag somewhere?

Reply:
you are right! I had a reset, but when I discovered that there was no track in the delete and started to rework it, I lost it